CMakeLists.txt

```diff
@@ -165,6 +165,23 @@ set(LINK_LIBRARY_LIST ${LINK_LIBRARY_LIST} Threads::Threads)
 # -std=gnu++14 -- Conform to the ISO 2014 C++ standard with GNU extensions.
 set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fPIC -mavx2 -mf16c -mfma -std=gnu++14")
 
+# Check AVX/SSE support
+include(CheckCXXCompilerFlag)
+
+# Check AVX flag
+check_cxx_compiler_flag("-mavx" COMPILER_SUPPORTS_AVX)
+if(COMPILER_SUPPORTS_AVX)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -mavx")
```

Comment:
-mavx2 is alreaddy to cmake flags at L166, i am under impression -mavx2 will include both avx and avx2
if that is the case, we might not require this flags addition again at L174

Reply:
Done